Document unavailable reduction ops with NCCL backend

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42822 Document unavailable reduction ops with NCCL backend**

These ops arent supported with NCCL backend and used to silently error.
We disabled them as part of addressing https://github.com/pytorch/pytorch/issues/41362, so
document that here.

Differential Revision: [D23023046](https://our.internmc.facebook.com/intern/diff/D23023046/)

rohan-varma committed:
Document unavailable reduction ops with NCCL backend

These ops arent supported with NCCL backend and used to silently error.
We disabled them as part of addressing https://github.com/pytorch/pytorch/issues/41362, so
document that here.

Differential Revision: [D23023046](https://our.internmc.facebook.com/intern/diff/D23023046/)

[ghstack-poisoned]

pritamdamania87 commented on torch/csrc/distributed/c10d/init.cpp:
```diff
@@ -186,6 +186,9 @@ PyObject* c10d_init(PyObject* _unused) {
 An enum-like class for available reduction operations: ``SUM``, ``PRODUCT``,
 ``MIN``, ``MAX``, ``BAND``, ``BOR``, and ``BXOR``.
 
+Note that ``BAND``, ``BOR``, and ``BXOR`` reductions are not available when
+using the ``NCCL`` backend.
+
```
Do we return an appropriate exception when these ops are used with NCCL? If so, can we add tests that validate this?

rohan-varma replied:
Yes, this was done as part of https://github.com/pytorch/pytorch/pull/42669/ and there are tests for the exception.

rohan-varma committed:
Update on "Document unavailable reduction ops with NCCL backend"


These ops arent supported with NCCL backend and used to silently error.
We disabled them as part of addressing https://github.com/pytorch/pytorch/issues/41362, so
document that here.

Differential Revision: [D23023046](https://our.internmc.facebook.com/intern/diff/D23023046/)

[ghstack-poisoned]
